crates/sui-core/src/unit_tests/gas_tests.rs

```diff
@@ -241,19 +244,16 @@ async fn test_native_transfer_insufficient_gas_execution() {
         .unwrap()
         .into_effects_for_testing()
         .into_data();
-    // We won't drain the entire budget because we don't charge for storage if tx failed.
-    assert!(effects.gas_cost_summary().gas_used() < budget);
+    // Transaction failed for out of gas so charge is same as budget
```

Comment:
@sblackshear see if this makes sense to you

Reply:
Yep!